### PR TITLE
Creating a parenthesis web service

### PR DESCRIPTION
**Pull request by Konstantsiy**

### CODE REVIEW

File: parentheses/service/service.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+	length, err := strconv.Atoi(n)
+	if err != nil || length <= 0 {
+		http.Error(w, ErrIncorrectLength.Error(), http.StatusBadRequest)
+	}
+
+	resultStr, err := generateRandomSequence(length)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
```
Comment (papahmsolo) [inline parentheses/service/service.go]: This code will write the error twice for an invalid `n`

File: parentheses/service/service.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+// generateRandomSequence generates the random sequence of brackets with the specified length.
+func generateRandomSequence(length int) (string, error) {
+	if length <= 0 {
+		return "", ErrIncorrectLength
+	}
+
+	rand.Seed(time.Now().UnixNano())
+	runes := make([]rune, length)
```
Comment (papahmsolo) [inline parentheses/service/service.go]: We're dealing with brackets only, so this could be a slice of bytes

File: parentheses/service/service.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	for i := 0; i < length; i++ {
+		runes[i] = rune(bracketsArray[rand.Intn(length)])
+	}
```
Comment (papahmsolo) [inline parentheses/service/service.go]: Index might be out of range here

**Review by papahmsolo — CHANGES_REQUESTED** [review]
Let's make some changes:

- `generateRandomSequence` belongs to the `brackets` package
- let's write some test for it
- how do we start the service? Let's also update the readme

File: parentheses/service/service.go (hunk 1 above)
Comment (Konstantsiy) [reply parentheses/service/service.go]: > This code will write the error twice for an invalid `n`

Sorry, I was in a hurry to push

File: parentheses/service/service.go (hunk 3 above)
Comment (Konstantsiy) [reply parentheses/service/service.go]: > Index might be out of range here

Sorry, my mistake

File: parentheses/brackets/brackets.go [hunk 4]
```diff
@@ -1,12 +1,23 @@
...
-	'[': ']',
-	'{': '}',
-}
+import (
+	"errors"
+	"math/rand"
+	"time"
+)
+
+// Package level Error.
```
Comment (papahmsolo) [inline parentheses/brackets/brackets.go]: Not sure about the meaning on this

File: parentheses/service/service.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+)
+
+// processRequest accepts a query with length parameter and
+// returns the random generated string of brackets with specified length.
+func processRequest(w http.ResponseWriter, r *http.Request) {
+	n := r.URL.Query().Get("n")
+
+	length, err := strconv.Atoi(n)
+	if err != nil || length <= 0 {
+		http.Error(w, "Incorrect request param of string length. Toy need a positive number", http.StatusBadRequest)
```
Comment (papahmsolo) [inline parentheses/service/service.go]: Toy Story? 😁

File: parentheses/service/service.go [hunk 6]
```diff
@@ -0,0 +1,41 @@
+// Package service implements a parentheses web service that
+// generates a random sequence of parentheses of the given length.
+package service
+
+import (
+	"ita/parentheses/brackets"
```
Comment (papahmsolo) [inline parentheses/service/service.go]: You'll need to import this by the mod name and package 
`"github.com/Konstantsiy/go-internship/parentheses/brackets"`

File: parentheses/service/service.go [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+	}
+}
+
+// Run starts starts handling requests.
+func Run() {
+	http.HandleFunc("/generate", processRequest)
+	if err := http.ListenAndServe(":8080", nil); err != nil {
+		log.Fatal(err)
+	}
+}
```
Comment (papahmsolo) [inline parentheses/service/service.go]: You can simply make this a `main` function and along with the package 👌

**Review by papahmsolo — CHANGES_REQUESTED** [review]
Looks good! Almost there.

File: parentheses/service/service.go (hunk 5 above)
Comment (Konstantsiy) [reply parentheses/service/service.go]: > Toy Story?

Yep :)

**Review by papahmsolo — APPROVED** [review]
Looks good👍

**Pull request merged**